boring-sys/build.rs

```diff
@@ -584,7 +584,19 @@ fn main() {
     println!("cargo:rerun-if-env-changed=BORING_SSL_PRECOMPILED_BCM_O");
     println!("cargo:rerun-if-env-changed=BORINGSSL_BUILD_DIR");
 
-    #[cfg(all(feature = "fips", feature = "rpk"))]
+    // When using the "fips" feature, boring-sys produces bindings for the entire FIPS-validated
+    // module. When using "fips-link-compiled", a newer version of boringSSL is built, but links
+    // implementations of certain algorithms from the FIPS module. These features are mutually
+    // exclusive. Furthermore, any feature that requires a newer version of boringSSL, or patching
+    // boringSSL, is incompatible with "fips".
+    #[cfg(all(
+        feature = "fips",
+        any(
+            feature = "fips-link-precompiled",
+            feature = "rpk",
+            feature = "pq-experimental"
+        )
+    ))]
```

Comment:
Compile error message needs an update

Reply:
Good catch, updated. Though I may be misunderstnding how things work, I think I need to do more digging to see how the "fips-link-precompiled" and "fips" are supposed to interact. (I assumed they were mutually exclusive but perhaps not?)